### PR TITLE
add 2 cluster delete apis

### PR DESCRIPTION
**Pull request by yingwang-us**
https://github.com/razee-io/Razeedash-api/issues/289

add follow 2 apis and test cases:

mutation {
  deleteClusterByClusterID(org_id: "0c27e376-a36d-4ee0-ba90-fed4ce7988b1"
  cluster_id: "4f1a4a82-879c-4f01-be87-0312eeec5893") {
    deletedClusterCount
    deletedResourceCount
  }
}

mutation {
  deleteClusters (org_id: "0c27e376-a36d-4ee0-ba90-fed4ce7988b1") {
    deletedClusterCount
    deletedResourceCount
  }
}

### CODE REVIEW

File: app/apollo/resolvers/cluster.js [hunk 1]
```diff
@@ -183,6 +183,65 @@ const clusterResolvers = {
...
+      const queryName = 'deleteClusters';
+      const { models, me, req_id, logger } = context;
+      logger.debug({req_id, user: whoIs(me), org_id}, `${queryName} enter`);
+
+      await validAuth(me, org_id, ACTIONS.MANAGE, TYPES.CLUSTER, queryName, context);
+
+      try {
+        const deletedClusters = await models.Cluster.deleteMany({ org_id });
+
+        const deletedResources = await models.Resource.updateMany({ org_id },
```
Comment (mckaymic) [inline app/apollo/resolvers/cluster.js:231]: If we delete a cluster we should just delete the resource as well (not soft delete)

File: app/apollo/resolvers/cluster.js [hunk 2]
```diff
@@ -183,6 +183,65 @@ const clusterResolvers = {
...
+      const { models, me, req_id, logger } = context;
+      logger.debug({req_id, user: whoIs(me), org_id, cluster_id}, `${queryName} enter`);
+
+      await validAuth(me, org_id, ACTIONS.MANAGE, TYPES.CLUSTER, queryName, context);
+
+      try {
+        const deletedCluster = await models.Cluster.findOneAndDelete({org_id,
+          cluster_id});
+
+        const deletedResources = await models.Resource.updateMany({ org_id, cluster_id },
```
Comment (mckaymic) [inline app/apollo/resolvers/cluster.js:203]: Delete the resources from the database (not soft delete)

File: app/apollo/schema/cluster.js [hunk 3]
```diff
@@ -44,6 +44,16 @@ const clusterSchema = gql`
     count: Int
   }
 
+  type DeleteClusterByClusterIDResponse {
```
[the remark this answers is not in the document]
Comment (yingwang-us) [reply app/apollo/schema/cluster.js]: removed DeleteClustersResponse.

File: app/apollo/resolvers/cluster.js (hunk 1 above)
Comment (yingwang-us) [reply app/apollo/resolvers/cluster.js:235]: per discussion with mike, keep soft delete for now until we have a background process to remove contents from COS.

File: app/apollo/resolvers/cluster.js (hunk 2 above)
Comment (yingwang-us) [reply app/apollo/resolvers/cluster.js:205]: per discussion with mike, keep soft delete for now until we have a background process to remove contents from COS.

**Pull request merged**